src/geom/path_builder.rs

```diff
@@ -0,0 +1,357 @@
+                    self.endpoint = point;
+                }
+                self.path.0.push(item);
+            }
+            pub fn extend(&mut self, item: PathBuilderItem<Abs>) {
+                let map_axes = |p: Axes<DeltaAbs<Rel<Abs>>>| Point {
+                    x: p.x.map(|l| l.relative_to(self.base.x)).to_abs(self.endpoint.x),
+                    y: p.y.map(|l| l.relative_to(self.base.y)).to_abs(self.endpoint.y),
+                };
+                match item {
+                    PathBuilderItem::MoveTo(p) => {
+                        self.add_item(PathItem::MoveTo(map_axes(p)))
+                    }
+                    PathBuilderItem::LineTo(p) => {
+                        self.add_item(PathItem::LineTo(map_axes(p)))
+                    }
+                    PathBuilderItem::CubicTo(p1, p2, p3) => {
+                        self.add_item(PathItem::CubicTo(
+                            map_axes(p1),
+                            map_axes(p2),
+                            map_axes(p3),
+                        ));
+                    }
+                    PathBuilderItem::ArcTo { to, radius, x_rotation, large, sweep } => {
+                        let arc = kurbo::SvgArc {
+                            from: self.endpoint.into(),
+                            to: map_axes(to).into(),
+                            radii: radius.into(),
+                            x_rotation: x_rotation.to_rad(),
+                            large_arc: large,
+                            sweep,
+                        };
+
+                        if arc.is_straight_line() {
+                            self.add_item(PathItem::MoveTo(arc.to.into()));
+                        } else {
+                            // Can only fail if `is_straight_line` returns true.
+                            kurbo::Arc::from_svg_arc(&arc).unwrap().to_cubic_beziers(
+                                1e-5,
+                                |p1, p2, p3| {
+                                    self.add_item(PathItem::CubicTo(
+                                        p1.into(),
+                                        p2.into(),
+                                        p3.into(),
+                                    ));
+                                },
+                            );
+                        }
+                    }
+                    PathBuilderItem::ClosePath => self.path.0.push(PathItem::ClosePath),
+                }
+            }
+        }
+
+        let mut builder = Builder {
+            path: Path(vec![PathItem::MoveTo(Point::zero())]),
+            endpoint: Point::zero(),
+            base,
+        };
+
+        for item in self.0.into_iter() {
+            builder.extend(item);
+        }
+
+        builder.path
+    }
+}
+
+/// A value that is either absolute or a delta of an
+/// earlier value.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum DeltaAbs<T> {
+    Abs(T),
+    Delta(T),
+}
```

Comment:
I am not completely in love with this enum yet, so better suggestions for name or an entirely different strategy would be welcome.